pkg/csi/service/cns/controller.go

```diff
@@ -251,6 +268,42 @@ func (c *controller) DeleteVolume(ctx context.Context, req *csi.DeleteVolumeRequ
 	if err != nil {
 		return nil, err
 	}
+	// Get the list of persistent volumes from k8s
+	pvList, err := c.pvLister.List(labels.Everything())
+	if err != nil {
+		msg := fmt.Sprintf("DeleteVolume: Error getting PersistentVolumes from API server with err: %v", err)
+		klog.Error(msg)
+		return nil, status.Errorf(codes.Internal, msg)
+	}
+	klog.V(4).Infof("DeleteVolume: pvList: %+v", pvList)
+	var pvFound bool
+	var pvName string
+	// Find the PV name for the delete volume request
+	for _, pv := range pvList {
+		if pv.Spec.CSI != nil {
+			if pv.Spec.CSI.VolumeHandle == req.VolumeId {
+				pvFound = true
+				pvName = pv.Name
+			}
+		}
+	}
+	vaList, err := c.vaLister.List(labels.Everything())
+	if err != nil {
+		msg := fmt.Sprintf("DeleteVolume: Error getting VolumeAttachments from API server with err: %v", err)
+		klog.Error(msg)
+		return nil, status.Errorf(codes.Internal, msg)
+	}
+	// After finding the PV name, check if the PV is found in any of the volume attachments from k8s
+	if pvFound {
```

Comment:
What if PV isn't found? Is it right to go ahead here..

Reply:
IMO, here we should proceed with DeleteVolume, i think. If the pv Lister(cache) could not list the PV for some reason, we should go ahead and delete the volume when we get a call in CSI controller.

Reply:
PV is getting deleted only after CSI returns sucess. So i think we should error out if driver can not find PV in the list.

Reply:
Addressed in the latest changes